core/src/main/java/org/elasticsearch/search/suggest/term/TermSuggestionBuilder.java

```diff
@@ -493,9 +526,18 @@ public StringDistanceImpl readFrom(final StreamInput in) throws IOException {
             return values()[ordinal];
         }
 
-        public static StringDistanceImpl fromString(final String str) {
+        public static StringDistanceImpl resolve(final String str) {
             Objects.requireNonNull(str, "Input string is null");
-            return valueOf(str.toUpperCase(Locale.ROOT));
+            try {
+                return valueOf(str.toUpperCase(Locale.ROOT));
+            } catch (IllegalArgumentException e) {
```

Comment:
I don't think raising en exception to save a few lines of code here is a good idea, please change this back to how it was before.